tenets/codelingo/code-review-comments/named-result-params/codelingo.yaml

```
@@ -0,0 +1,68 @@
+funcs:
+  - name: isSameName
+    type: asserter
+    body: |
+      function(a, b) {
+        return a.toLowerCase() === b.toLowerCase()
+      }
+tenets:
+  - name: named-result-params
+    flows:
+      codelingo/review:
+        comment: Consider what it will look like in godoc. {{returnName}} {{returnType}} will stutter in godoc; better to use {{returnType}}. From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters)
```

```suggestion
        comment: Avoid result params with names that are the same as their type. From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters)
```

I wasn't able to understand how godoc stutters - I didn't manage to run it, so I couldn't verify if your stuttering sentence makes sense. Can you clarify? If you basically copy pasted, then let's just rely on the link.

---

@BlakeMScurr Yeah that was copy/pasted from the doc. I assume it just means it looks silly. This is output of godoc if curious:

```
PACKAGE DOCUMENTATION

package somepackage
    import "somepackage"


TYPES

type Example struct {
    Count    int
    TheThing *Thing
}

    Example is an example

func (e *Example) GetAll() (thing Thing, count int)
    GetAll returns all the Thing

func (e *Example) GetCount() (count int)
    GetCount returns the count field of the Thing

func (e *Example) GetThing() (thing *Thing)
    GetThing returns the Example's Thing

func (e *Example) GetThingAsValue() (thing Thing)
    GetThingAsValue returns the Example's Thing as value

type Thing struct {
    Name   string
    Number int
}

    Thing is an example

func Extract(e *Example) (thing Thing)
    Extract a Thing

```

---

@Daanikus Huh, I assumed it meant that the sentence would be long and unwieldy, but it doesn't look like it even includes the type. I don't think it looks that silly. What you do you find silly about it?

How did you you generate this btw?

---

@BlakeMScurr I don't necessarily find it silly, I just assume that's what the spec is referring to. I think I've seen it mentioned before about having named returns sharing the name with their type and to avoid it.

Doc was generated by running `godoc somepackage`. It's gotta be in a dir with the same name as package and be in GOPATH.